_blog/2018/egress-tcp.md

```diff
@@ -0,0 +1,306 @@
+Created config egress-rule/default/mysql at revision 1954425
+```
+Note that for a TCP egress rule, we specify `tcp` as the protocol of a port of the rule. Also note that we use an IP of the external service instead of its domain name. I will talk more about TCP Egress Rules [below](#egress-rules-for-tcp-traffic). For now, let's verify that the egress rule we added fixed the problem. Let's access the webpage and see if the stars are back.
+
+It worked! Accessing the web page of the application displays the ratings without error:
+
+{% include figure.html width='80%' ratio='36.69%'
+    img='./img/externalMySQLRatings.png'
+    alt='Book Ratings Displayed Correctly'
+    title='Book Ratings Displayed Correctly'
+    caption='Book Ratings Displayed Correctly'
+    %}
+
+Note that we see a one-star rating for the both displayed reviews, as expected. I changed the ratings to be one star to provide us a visual clue that our external database is indeed used.
+
+As with egress rules for HTTP/HTTPS, we can delete and create egress rules for TCP using `istioctl`, dynamically.
+
+## Motivation for egress TCP traffic control
+Some in-mesh Istio applications must access external services, for example legacy systems. In many cases, the access is not performed over HTTP or HTTPS protocols. Other TCP protocols are used, for example database specific protocols [MongoDB Wire Protocol](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/) and [MySQL CLient/Server Protocol](https://dev.mysql.com/doc/internals/en/client-server-protocol.html) to communicate with external databases.
+
+Note that in case of access to external HTTPS services, as described in the [Control Egress TCP Traffic]({{home}}/docs/tasks/traffic-management/egress.html) task, an application must issue HTTP requests to the external service. The Envoy sidecar proxy attached to the pod or the VM, will intercept the requests and will open an HTTPS connection to the external service. The traffic will be unencrypted inside the pod or the VM, but it will leave the pod or the VM encrypted.
+
+However, sometimes this approach cannot work due to the following reasons:
+* The code of the application is configured to use an HTTPS URL and cannot be changed
+* The code of the application uses some library to access the external service and that library uses HTTPS only
+* There are compliance requirements that do not allow unencrypted traffic, even if the traffic is unencrypted only inside the pod or the VM
+
+In this case, HTTPS can be treated by Istio as _opaque TCP_ and can be handled in the same way as other TCP non-HTTP protocols.
+
+Next let's see how we define egress rules for TCP traffic.
+
+## Egress rules for TCP traffic
+The egress rules for enabling TCP traffic to a specific port must specify `TCP` as the protocol of the port. Additional non-HTTP TCP protocol currently supported is `MONGO`, the [MongoDB Wire Protocol](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/).
```

Comment:
This is not clear ... does this mean that the egress rule must specify `TCP` for the port protocol OR, in the case of mongodb, it can (or must) set the port protocol to `MONGO`?

Reply:
egress rules can specify either `TCP` for the port protocol, or (can, not must) specify `MONGO` in case of mongodb.

Reply:
How about rephrasing like this:

Additionally, for the [MongoDB Wire Protocol](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/), the protocol can be specified as `MONGO`, instead of `TCP`.

Reply:
Done, thanks!